airbyte/mcp/cloud_ops.py

```diff
@@ -686,6 +687,189 @@ def permanently_delete_custom_source_definition(
     )
 
 
+def update_cloud_source_config(
+    source_id: Annotated[
+        str,
+        Field(description="The ID of the deployed source to update."),
+    ],
+    *,
+    name: Annotated[
+        str | None,
+        Field(
+            description="Optional new name for the source.",
+            default=None,
+        ),
+    ] = None,
+    config: Annotated[
+        dict | str | None,
+        Field(
+            description="Optional new configuration for the source connector.",
+            default=None,
+        ),
+    ] = None,
+    config_secret_name: Annotated[
+        str | None,
+        Field(
+            description="The name of the secret containing the configuration.",
+            default=None,
+        ),
+    ] = None,
+) -> str:
+    """Update a deployed source connector's configuration on Airbyte Cloud.
+
+    This is a destructive operation that can break existing connections if the
+    configuration is changed incorrectly. Use with caution.
+
+    By default, the `AIRBYTE_CLIENT_ID`, `AIRBYTE_CLIENT_SECRET`, `AIRBYTE_WORKSPACE_ID`,
+    and `AIRBYTE_API_ROOT` environment variables will be used to authenticate with the
+    Airbyte Cloud API.
+    """
+    try:
+        workspace: CloudWorkspace = _get_cloud_workspace()
+        source = workspace.get_source(source_id=source_id)
+
+        config_dict = None
+        if config is not None or config_secret_name is not None:
+            config_dict = resolve_config(
+                config=config,
+                config_secret_name=config_secret_name,
+                config_spec_jsonschema=None,  # We don't have the spec here
+            )
+
+        source.update_config(name=name, config=config_dict)
+
+    except Exception as ex:
+        return f"Failed to update source '{source_id}': {ex}"
+    else:
+        return f"Successfully updated source '{source_id}'. " f"URL: {source.connector_url}"
+
+
+def update_cloud_destination_config(
```

Comment:
All the feedback about decoupling "update_config" from "rename" and "update_schedule" applies also to the tools.